### PR TITLE
Add algo for mapping external data to ndef, fixes #324

### LINKED ISSUE #324: Missing descriptions on pushing "external type" records
Here: https://w3c.github.io/web-nfc/#creating-web-nfc-message

@kenchris @zolkis @riju

### PR DESCRIPTION
**Pull request by kenchris**
***
<a href="https://pr-preview.s3.amazonaws.com/kenchris/web-nfc/pull/326.html" title="Last updated on Sep 2, 2019, 9:32 AM UTC (5f95fab)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/web-nfc/326/6ab55b8...kenchris:5f95fab.html" title="Last updated on Sep 2, 2019, 9:32 AM UTC (5f95fab)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2888,11 +2900,59 @@ <h3><dfn>Writing or pushing content</dfn></h3>
...
+          </li>
+          <li>
+            Let |ndefRecord| be the notation for the <a>NDEF record</a> to
+            be created by the UA.
+            <ol>
+              <li>
+                Set |ndefRecord|'s <a>TNF field</a> to `4` (<a>external type record</a>).
+              </li>
+              <li>
+                Set the |ndefRecord|'s <a>TYPE field</a> to |record|'s recordType.
```
Comment (zolkis) [inline index.html:2934]: I guess this is tricky, too. No checks needed?

File: index.html [hunk 2]
```diff
@@ -2888,11 +2900,59 @@ <h3><dfn>Writing or pushing content</dfn></h3>
       </p>
       </section>
 
+      <section><h3>Mapping external data to NDEF</h3>
```
Comment (zolkis) [inline index.html:2903]: I guess user data would be more appropriate than "external" data. The work external is a bit overloaded in this context.

**Review by zolkis — COMMENTED** [review]
Minor comments. Thanks for the work.

File: index.html (hunk 1 above)
Comment (kenchris) [reply index.html:2934]: if it gets here it should follow the ABNF, so all should be correct

File: index.html (hunk 2 above)
Comment (kenchris) [reply index.html:2903]: I thought about user or custom, but I actually find this more clear

**Pull request merged**